### PR TITLE
Add inverse error function

### PR DESCRIPTION
**Pull request by BaxterEaves**
I added everything to the `error` file, so let me know if you'd like things to be organized differently.

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -1,4 +1,7 @@
 use math;
+use std::f64;
```
Comment (IvanUkhov) [inline src/error.rs]: What is the purpose of this import?

File: src/error.rs [hunk 2]
```diff
@@ -1,4 +1,7 @@
...
+
+const SQRT_PI: f64 = 1.772453850905515881919427556567825376987457275391f64;
```
Comment (IvanUkhov) [inline src/error.rs]: Is the suffix needed in this case?

File: src/error.rs [hunk 3]
```diff
@@ -7,10 +10,53 @@ pub trait Error {
...
+        p = 0.00573950773 + p * w;
+        p = -0.0076224613 + p * w;
+        p = 0.00943887047 + p * w;
+        p = 1.00167406 + p * w;
+        p = 2.83297682 + p * w;
+    }
+
+    let res_ra = p * z; // assign to rational estimate variable
+
+    // Halley's method to refine estimate of inverse erf
```
Comment (IvanUkhov) [inline src/error.rs]: Let’s keep the comment before the function and remove the ones inside.

File: src/error.rs [hunk 4]
```diff
@@ -7,10 +10,53 @@ pub trait Error {
...
+    let fx = unsafe { math::erf(res_ra) } - z;
+    let df = 2.0 / SQRT_PI * (-(res_ra * res_ra)).exp();
+    let d2f = -2.0 * res_ra * df;
+
+    res_ra - (2.0 * fx * df) / ((2.0 * df * df) - (fx * d2f))
 }
 
 macro_rules! implement {
-    ($kind:ty) => (
+    ($kind:ty) => {
```
Comment (IvanUkhov) [inline src/error.rs]: Why did this have to be changed?

File: src/error.rs [hunk 5]
```diff
@@ -21,9 +67,46 @@ macro_rules! implement {
             fn erfc(self) -> Self {
                 unsafe { math::erfc(self as f64) as Self }
             }
+
+            #[inline]
+            fn erfinv(self) -> Self {
+                erfinv(self as f64) as Self
```
Comment (IvanUkhov) [inline src/error.rs]: My main concern is that this conversion. There should be two functions generated: one for `f32` and one for `f64`. Have a look at `beta.rs`, for instance.

File: src/error.rs [hunk 6]
```diff
@@ -21,9 +67,46 @@ macro_rules! implement {
...
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use assert;
+
+    // Test values calculated from scipy
+    #[test]
+    fn erfinv_of_zero_should_be_zero() {
+        assert::close(0.0_f64.erfinv(), 0.0, 1E-12);
```
Comment (IvanUkhov) [inline src/error.rs]: Let’s be consistent and use the lowercase _e_ here and below. And why do you need those suffixes?

File: src/error.rs (hunk 2 above)
Comment (IvanUkhov) [reply src/error.rs]: Surprised that `std::f64::consts` doesn’t have it.

File: src/error.rs (hunk 4 above)
Comment (BaxterEaves) [reply src/error.rs]: Rustfmt did that. Would you prefer them to be parens?

File: src/error.rs (hunk 2 above)
Comment (BaxterEaves) [reply src/error.rs]: The suffix is not needed, but that is the convention used in `std::f64::consts`. Happy to remove it.

**Comment by IvanUkhov** [comment]
Thank you!

**Comment by BaxterEaves** [comment]
My pleasure. Thanks for the speedy and informative review!

**Pull request merged**